### DISCUSSION TITLE
Retained windows

### DISCUSSION THREAD

**Discussion by pamoroso** [Interlisp/medley#1748] [Q&A]
In section "4.4 INTERLISP-D" of the paper [Ten Years of Window Systems - A Retrospective View](https://www.chilton-computing.org.uk/inf/literature/books/wm/p004.htm) Warren Teitelman wrote:

> **Retained (or cached) windows** were added, initially for speeding up performance, for example with menus, but later retained windows were found useful for hiding from the programmer the fact that his window was not on top.

What are retained windows? Did Teitelman mean ordinary windows (e.g. created with `CREATEW`) or some specific types?

**Comment by nbriggs** [comment]
I believe he is talking about the implementation where a window retains the image that it is obscuring, so when it closes it restores the part of the screen underneath. A window only gets drawn on when it is on top, and the bits go directly to the screen bitmap Sent from my iPhoneOn Jun 5, 2024, at 06:35, Paolo Amoroso ***@***.***> wrote:﻿
In section "4.4 INTERLISP-D" of the paper Ten Years of Window Systems - A Retrospective View Warren Teitelman wrote:

Retained (or cached) windows were added, initially for speeding up performance, for example with menus, but later retained windows were found useful for hiding from the programmer the fact that his window was not on top.

What are retained windows? Did Teitelman mean ordinary windows (e.g. created with CREATEW) or some specific types?
